Build regression after upgrading MySQL version on Windows

# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/sourcemod/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Operating System version:  Microsoft Windows Server 2022 10.0.20348
  * Game/AppID (with version if applicable): 
  * Current SourceMod version: 
  * Current SourceMod snapshot: 
  * Current Metamod: Source snapshot:
  - [ ] I have updated SourceMod to the [latest version](https://www.sourcemod.net/downloads.php) and it still happens.
  - [ ] I have updated SourceMod to the [latest snapshot](https://www.sourcemod.net/downloads.php?branch=dev) and it still happens.
  - [ ] I have updated SourceMM to the [latest snapshot](https://sourcemm.net/downloads.php?branch=dev) and it still happens.

# Description
#2192 broke building the mysql extension on Windows - but only after some time? Probably Github Actions caching the old MySQL version?
The [release notes](https://downloads.mysql.com/docs/mysql-5.5-relnotes-en.pdf) for MySQL 5.5.55 state under `Compilation Notes`:
> Windows builds now use the default runtime libraries (builds use the `/MD` flag). (Bug #25611609)

Which was the reason 5.5.54 was selected on Windows. We compile statically using `/MT` which links the libcmt.lib common runtime statically. The mysqlclient links the dynamic .dll now too which causes a conflict and would require the dynamic library to be available in the library path when loading the mysql extension.

This causes builds to fail to link in CI. The build servers don't use the checkout-deps.sh script and thus are still on the old mysql version 5.5.54.


# Problematic Code (or Steps to Reproduce)
  ```
Build sourcemod on Windows and grab dependencies using the `tools/checkout-deps.sh` script.
  ```

# Logs
```
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library

LINK : warning LNK4286: symbol 'exit' defined in 'libucrt.lib(exit.obj)' is imported by 'mysqlclient.lib(str_alloc.obj)'

...


mysqlclient.lib(my_init.obj) : error LNK2001: unresolved external symbol __imp_getenv

mysqlclient.lib(handshake_client.obj) : error LNK2001: unresolved external symbol __imp_getenv

...
```
https://github.com/alliedmodders/sourcemod/actions/runs/12089683390/job/33715394830

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -33,7 +33,7 @@ jobs:
       ARCH: x86,x86_64
       DEPENDENCIES_FOLDER: dependencies
       DEPENDENCIES_ROOT: ${{ github.workspace }}/dependencies
-      MYSQL_VERSION: '5.5'
+      MYSQL_VERSION: '5.7'
       MMSOURCE_VERSION: '1.12'
     steps:
       - uses: actions/checkout@v4
@@ -69,7 +69,7 @@ jobs:
 
           # Satisfy checkout-deps requirement for a "sourcemod" folder.
           mkdir -p sourcemod
-          ../sourcemod/tools/checkout-deps.sh -s ${{ join(fromJSON(env.SDKS)) }}
+          ../sourcemod/tools/checkout-deps.sh -s ${{ join(fromJSON(env.SDKS)) }} -d
 
       - name: Install Linux dependencies
         if: startsWith(runner.os, 'Linux')

--- a/AMBuildScript
+++ b/AMBuildScript
@@ -146,7 +146,7 @@ class SMConfig(object):
           self.mysql_root['x86'] = builder.options.mysql_path
         else:
           for i in range(10):
-            self.mysql_root['x86'] = ResolveEnvPath('MYSQL55', 'mysql-5.' + str(i))
+            self.mysql_root['x86'] = ResolveEnvPath('MYSQL57', 'mysql-5.' + str(i))
             if self.mysql_root['x86']:
               break
         if not self.mysql_root['x86'] or not os.path.isdir(self.mysql_root['x86']):
@@ -158,7 +158,7 @@ class SMConfig(object):
           self.mysql_root['x86_64'] = builder.options.mysql64_path
         else:
           for i in range(10):
-            self.mysql_root['x86_64'] = ResolveEnvPath('MYSQL55_64', 'mysql-5.' + str(i) + '-x86_64')
+            self.mysql_root['x86_64'] = ResolveEnvPath('MYSQL57_64', 'mysql-5.' + str(i) + '-x86_64')
             if self.mysql_root['x86_64']:
               break
         if not self.mysql_root['x86_64'] or not os.path.isdir(self.mysql_root['x86_64']):

--- a/extensions/mysql/AMBuilder
+++ b/extensions/mysql/AMBuilder
@@ -13,6 +13,10 @@ if SM.mysql_root:
       binary.compiler.cxxflags += ['-fno-rtti']
     elif binary.compiler.family == 'msvc':
       binary.compiler.cxxflags += ['/GR-']
+      if builder.options.debug == '1':
+        binary.compiler.cflags += ['/MDd']
+      else:
+        binary.compiler.cflags += ['/MD']
 
     if binary.compiler.target.platform == 'linux' or binary.compiler.target.platform == 'mac':
       binary.compiler.postlink += [
@@ -25,8 +29,17 @@ if SM.mysql_root:
       binary.compiler.postlink += ['-lrt']
     elif binary.compiler.target.platform == 'windows':
       binary.compiler.defines += ['WIN32_LEAN_AND_MEAN']
+      if builder.options.debug == '1':
+        binary.compiler.defines += ['_ITERATOR_DEBUG_LEVEL=2']
+        binary.compiler.postlink += [
+          os.path.join(SM.mysql_root[arch], 'lib', 'debug', 'mysqlclient.lib'),
+        ]
+      else:
+        binary.compiler.postlink += [
+          os.path.join(SM.mysql_root[arch], 'lib', 'mysqlclient.lib'),
+        ]
       binary.compiler.postlink += [
-        os.path.join(SM.mysql_root[arch], 'lib', 'mysqlclient.lib'),
+        'crypt32.lib',
         'wsock32.lib'
       ]
 

--- a/tools/checkout-deps.sh
+++ b/tools/checkout-deps.sh
@@ -4,16 +4,21 @@
 trap "exit" INT
 
 download_mysql=1
+download_mysql_debug=1
 
 # List of HL2SDK branch names to download.
 # ./checkout-deps.sh -s tf2,css
 # Disable downloading of mysql libraries.
 # ./checkout-deps.sh -m
+# Disable downloading of mysql debug libraries on Windows.
+# ./checkout-deps.sh -d
 while getopts ":s:m" opt; do
   case $opt in
     s) IFS=', ' read -r -a sdks <<< "$OPTARG"
     ;;
     m) download_mysql=0
+    ;;
+    d) download_mysql_debug=0
     ;;
     \?) echo "Invalid option -$OPTARG" >&2
     ;;
@@ -59,13 +64,14 @@ getmysql ()
 }
 
 # 32-bit MySQL
-mysqlfolder=mysql-5.5
+mysqlfolder=mysql-5.7
 if [ $ismac -eq 1 ]; then
+  mysqlfolder=mysql-5.5
   mysqlver=mysql-5.5.28-osx10.5-x86
   mysqlurl=https://cdn.mysql.com/archives/mysql-5.5/$mysqlver.$archive_ext
 elif [ $iswin -eq 1 ]; then
-  mysqlver=mysql-5.5.62-win32
-  mysqlurl=https://cdn.mysql.com/archives/mysql-5.5/$mysqlver.$archive_ext
+  mysqlver=mysql-5.7.44-win32
+  mysqlurl=https://cdn.mysql.com/archives/mysql-5.7/$mysqlver.$archive_ext
   # The folder in the zip archive does not contain the substring "-noinstall", so strip it
   mysqlver=${mysqlver/-noinstall}
 else
@@ -77,19 +83,36 @@ if [ $download_mysql -eq 1 ]; then
 fi
 
 # 64-bit MySQL
-mysqlfolder=mysql-5.5-x86_64
+mysqlfolder=mysql-5.7-x86_64
 if [ $ismac -eq 1 ]; then
+  mysqlfolder=mysql-5.5-x86_64
   mysqlver=mysql-5.5.28-osx10.5-x86_64
   mysqlurl=https://cdn.mysql.com/archives/mysql-5.5/$mysqlver.$archive_ext
 elif [ $iswin -eq 1 ]; then
-  mysqlver=mysql-5.5.62-winx64
-  mysqlurl=https://cdn.mysql.com/archives/mysql-5.5/$mysqlver.$archive_ext
+  mysqlver=mysql-5.7.44-winx64
+  mysqlurl=https://cdn.mysql.com/archives/mysql-5.7/$mysqlver.$archive_ext
 else
   mysqlver=mysql-5.7.44-linux-glibc2.12-x86_64
   mysqlurl=https://cdn.mysql.com/archives/mysql-5.7/$mysqlver.$archive_ext
 fi
 if [ $download_mysql -eq 1 ]; then
   getmysql
+fi
+
+if [ $iswin -eq 1 && $download_mysql_debug -eq 1 ]; then
+  mysqlfolder=mysql-5.7-debug
+  mysqlver=mysql-5.7.44-win32
+  mysqlurl=https://cdn.mysql.com/archives/mysql-5.7/$mysqlver-debug-test.$archive_ext
+  getmysql
+  cp -r $mysqlfolder/lib/* mysql-5.7/lib
+  rm -rf $mysqlfolder
+
+  mysqlfolder=mysql-5.7-debug-x86_64
+  mysqlver=mysql-5.7.44-winx64
+  mysqlurl=https://cdn.mysql.com/archives/mysql-5.7/$mysqlver-debug-test.$archive_ext
+  getmysql
+  cp -r $mysqlfolder/lib/* mysql-5.7-x86_64/lib
+  rm -rf $mysqlfolder
 fi
 
 checkout ()

